conjure-api/src/main/conjure/conjure-api.yml

```diff
@@ -173,13 +173,19 @@ types:
           auth: optional<AuthType>
           args: list<ArgumentDefinition>
           returns: optional<Type>
+          errors: list<EndpointError>
           docs: optional<Documentation>
           deprecated: optional<Documentation>
           markers: list<Type>
           tags: set<string>
       EndpointName:
         alias: string
         docs: Should be in lowerCamelCase.
+      EndpointError:
+        fields:
+          error: TypeName
```

Comment:
I think TypeName may be a bit misleading here, there's a fair bit of overlap, but I think perhaps we should flatten the `TypeName` components (package + name) into this type, and name it `ErrorTypeName`